src/server/gc_worker/gc_worker.rs

```diff
@@ -422,6 +432,96 @@ where
         Ok((handled_keys, wasted_keys))
     }
 
+
+    fn raw_gc_keys(
+        &mut self,
+        keys: Vec<Key>,
+        safe_point: TimeStamp,
+  //      regions_provider: Option<(u64, Arc<dyn RegionInfoProvider>)>,
+    ) -> Result<(usize, usize)> {
+        let mut mvcc_snapshot
+            = RawBasicSnapshot::from_snapshot( self.engine.snapshot(Default::default()).unwrap());
+
+        let mut raw_modifys =MvccRaw::new();
+
+        let mut keys_iter = keys.iter();
+        let mut next_gc_key = keys_iter.next();
+        while let Some(ref key) = next_gc_key {
+            if let Err(e) = self.raw_gc_key(safe_point, key, &mut raw_modifys, &mut mvcc_snapshot) {
+                GC_KEY_FAILURES.inc();
+                error!(?e; "Raw GC meets failure"; "key" => %key,);
+                // Switch to the next key if meets failure.
+            }
+
+            //flush_raw_gc;
+            Self::flush_raw_gc(raw_modifys, &self.limiter, &self.engine)?;
+            //flush 之后重置rwModify=let RawModify=MvccRaw::new();
+            raw_modifys =MvccRaw::new();
+            next_gc_key = keys_iter.next();
+        }
+        Ok((0, 0))
+    }
+
+    fn raw_gc_key(
+        &mut self,
+        safe_point: TimeStamp,
+        key: &Key,
+        //  gc_info: &mut GcInfo,
+        raw_modifys: &mut MvccRaw,
+        snapshot: &mut RawBasicSnapshot<E::Snap>,
+    ) -> Result<()> {
+
+        let start_key = Key::from_raw(key.as_encoded()).append_ts(safe_point);
+        //let startKey = key.clone().append_ts(safe_point);
+
+        let mut cursor = CursorBuilder::new(snapshot, CF_DEFAULT)
+            .build()?;
+
+        let mut statistics= CfStatistics::default();
+
+        cursor.seek(&start_key, &mut statistics)?;
+
+        while cursor.valid()? {
+
+            let temp_key = Key::from_encoded_slice(cursor.key(&mut statistics));
+
+            let (usekey,userts) = APIV2::decode_raw_key(&temp_key, true)?;
+            let ts=userts.unwrap();
+            if ts==safe_point {
```

Comment:
How about set cursor lower bound to be `safe_point - 1` ? Then we don't need the judgement here.